processor/resourcedetectionprocessor/internal/azure/aks/aks_test.go

```diff
@@ -34,6 +34,8 @@ func TestDetector_Detect_K8s_Azure(t *testing.T) {
 	assert.Equal(t, map[string]any{
 		"cloud.provider": "azure",
 		"cloud.platform": "azure_aks",
+		// Cluster name will not be detected if resource grup is not set
+		"k8s.cluster.name": "",
```

Comment:
If the attribute is not enabled, it won't be available in the output, so you want to remove this. See failing test [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/7504616364/job/20432104381?pr=29328). 
I think it would be better to add another test case with the attribute enabled.